### PR TITLE
Merge feature/milestone-04-category-filter to develop branch

### PR DESCRIPTION
**Pull request by temesghentekeste**
Hi :hand:,

Milestone 04 is developed by creating a feature branch named  'feature/milestone-04-category-filter'.

As per the requirements, I: 
- [ ] Enhanced the Redux store by adding a new state named filter and add a new filter reducer
- [ ] Implemented the CategoryFilter component
- [ ] Kept the filter value in the Redux store

Kindly request a pull request before I merge my feature branch to the develop branch.

Regards,

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello  @temesghentekeste 👋

**You've done pretty well here, filter action and reducer have both been added. 👌**

However, let's make a couple of changes.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/containers/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  'Learning',
+  'Sci-Fi',
+];
+
+const CategoryFilter = ({ filterChange }) => {
+  const [category, setCategory] = useState('All');
+  const handleCategoryChange = (e) => {
+    filterChange(e.target.value);
+    setCategory(e.target.value);
+  };
```
Comment (mikenath223) [inline src/containers/CategoryFilter.js]: - [ ] Please move this component to the `src/components` folder as it is a presentational component placed in `src/components` directory [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/presentationalCategoryFilter.png)

Comment (temesghentekeste) [reply src/containers/CategoryFilter.js]: Thanks, @mikenath223 for your feedback; I have just addressed the issue.

Kindly request a pull request before I merge the feature branch to the develop branch.
Regarss,
Temesghen T Bahta

**Review by addod19 — APPROVED** [review]
## Status: Approved

Hi @temesghentekeste, congratulations on this milestone :+1: The categoryFilter has been moved into the "components" directory as mentioned in the previous review 😄 I hereby approved the project.

Kindly go ahead and merge into development, do not forget to submit the project completion form
_Happy coding_ 👨‍💻 
[Daniel Larbi Addo](https://www.linkedin.com/mwlite/in/daniel-larbi-addo)

**Comment by temesghentekeste** [comment]
Thanks, @addod19 for approving the project; I guess this one is the easiest review :smile: you ever got.

**Pull request merged**